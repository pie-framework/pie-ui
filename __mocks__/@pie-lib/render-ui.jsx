import React from 'react';

export const Feedback = () => <div>Feedback</div>;

export const indicators = {
  Correct: () => <div>Correct</div>,
};

export const PreviewPrompt = () => <div>PreviewPrompt</div>;

export const color = {
  text: jest.fn(),
  correct: jest.fn(),
  incorrect: jest.fn(),
  disabled: jest.fn(),
  primary: jest.fn(),
  primaryLight: jest.fn(),
  primaryDark: jest.fn(),
  primaryText: jest.fn(),
  secondary: jest.fn(),
  secondaryLight: jest.fn(),
  secondaryDark: jest.fn(),
  secondaryText: jest.fn(),
  background: jest.fn(),
};
