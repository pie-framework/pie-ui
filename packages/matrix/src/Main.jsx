import React from 'react';
import PropTypes from 'prop-types';
import {PreviewLayout} from '@pie-lib/render-ui';
import Matrix from './Matrix';

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func
  };

  static defaultProps = {
    model: {},
    session: {},
  };

  render() {
    const {model, onSessionChange, session} = this.props;
    return (
      <PreviewLayout>
        <Matrix
          {...model}
          session={session}
          onSessionChange={onSessionChange}
        />
      </PreviewLayout>
    );
  }
}

export default Main;
