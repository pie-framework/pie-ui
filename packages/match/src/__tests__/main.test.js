import * as React from 'react';
import Main from '../main';
import { shallowChild } from '@pie-lib/test-utils';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { Feedback } from '@pie-lib/render-ui';
import AnswerGrid from '../answer-grid';
import { shallow } from 'enzyme/build';
import Controls from '@pie-ui/graph-lines/src/controls';

describe('Main', () => {
  const defaultProps = {
    model: {
      config: {
        id: '1',
        element: 'match-element',
        rows: [{
          id: 1,
          title: 'Question Text 1',
          values: [false, false]
        }, {
          id: 2,
          title: 'Question Text 2',
          values: [false, false]
        }, {
          id: 3,
          title: 'Question Text 3',
          values: [false, false]
        }, {
          id: 4,
          title: 'Question Text 4',
          values: [false, false]
        }],
        shuffled: false,
        partialScoring: [],
        layout: 3,
        headers: ['Column 1', 'Column 2', 'Column 3'],
        choiceMode: 'radio',
        feedback: {
          correct: {
            type: 'none',
            default: 'Correct'
          },
          partial: {
            type: 'none',
            default: 'Nearly'
          },
          incorrect: {
            type: 'none',
            default: 'Incorrect'
          }
        },
      }
    },
    onSessionChange: jest.fn(),
    session: {}
  };

  let wrapper;
  let component;

  beforeEach(() => {
    wrapper = shallowChild(Main, defaultProps, 1);
  });

  describe('render', () => {
    let w;

    beforeEach(() => {
      w = props => shallow(<Main { ...props } />);
    });

    it('snapshot', () => {
      expect(w(defaultProps)).toMatchSnapshot();
    });

    it('snapshot with rationale', () => {
      expect(w({
        ...defaultProps,
        rationale: 'This is rationale'
      })).toMatchSnapshot();
    });

    it('snapshot with teacher Instructions', () => {
      expect(w({
        ...defaultProps,
        teacherInstructions: 'These are teacher instructions'
      })).toMatchSnapshot();
    });

    it('renders correctly', () => {
      component = wrapper();

      expect(component.find(CorrectAnswerToggle).length).toEqual(1);
      expect(component.find(Feedback).length).toEqual(0);
      expect(component.find(AnswerGrid).length).toEqual(1);

      expect(component.state()).toEqual({
        session: {
          answers: {
            1: [false, false],
            2: [false, false],
            3: [false, false],
            4: [false, false]
          }
        },
        showCorrect: false,
        shuffledRows: [{
          id: 1,
          title: 'Question Text 1',
          values: [false, false]
        }, {
          id: 2,
          title: 'Question Text 2',
          values: [false, false]
        }, {
          id: 3,
          title: 'Question Text 3',
          values: [false, false]
        }, {
          id: 4,
          title: 'Question Text 4',
          values: [false, false]
        }]
      })
    });
  });

  it('generates answers correctly from rows', () => {
    component = wrapper();

    expect(component.instance().generateAnswers({
      config: {
        layout: 3,
        rows: [{
          id: 1,
          title: 'Question Text 1',
          values: [true, false]
        }, {
          id: 4,
          title: 'Question Text 2',
          values: [false, true]
        }, {
          id: 12,
          title: 'Question Text 3',
          values: [false, false]
        }, {
          id: 9,
          title: 'Question Text 4',
          values: [true, true]
        }]
      }
    })).toEqual({
      1: [false, false],
      4: [false, false],
      12: [false, false],
      9: [false, false]
    })
  });
});
