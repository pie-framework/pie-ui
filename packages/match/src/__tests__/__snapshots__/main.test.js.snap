// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render snapshot 1`] = `
<Main
  classes={
    Object {
      "collapsible": "Main-collapsible-5",
      "main": "Main-main-2",
      "mainContainer": "Main-mainContainer-1",
      "prompt": "Main-prompt-4",
      "toggle": "Main-toggle-3",
    }
  }
  model={
    Object {
      "config": Object {
        "choiceMode": "radio",
        "element": "match-element",
        "feedback": Object {
          "correct": Object {
            "default": "Correct",
            "type": "none",
          },
          "incorrect": Object {
            "default": "Incorrect",
            "type": "none",
          },
          "partial": Object {
            "default": "Nearly",
            "type": "none",
          },
        },
        "headers": Array [
          "Column 1",
          "Column 2",
          "Column 3",
        ],
        "id": "1",
        "layout": 3,
        "partialScoring": Array [],
        "rows": Array [
          Object {
            "id": 1,
            "title": "Question Text 1",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 2,
            "title": "Question Text 2",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 3,
            "title": "Question Text 3",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 4,
            "title": "Question Text 4",
            "values": Array [
              false,
              false,
            ],
          },
        ],
      },
    }
  }
  onSessionChange={[MockFunction]}
  session={Object {}}
/>
`;

exports[`Main render snapshot with rationale 1`] = `
<Main
  classes={
    Object {
      "collapsible": "Main-collapsible-5",
      "main": "Main-main-2",
      "mainContainer": "Main-mainContainer-1",
      "prompt": "Main-prompt-4",
      "toggle": "Main-toggle-3",
    }
  }
  model={
    Object {
      "config": Object {
        "choiceMode": "radio",
        "element": "match-element",
        "feedback": Object {
          "correct": Object {
            "default": "Correct",
            "type": "none",
          },
          "incorrect": Object {
            "default": "Incorrect",
            "type": "none",
          },
          "partial": Object {
            "default": "Nearly",
            "type": "none",
          },
        },
        "headers": Array [
          "Column 1",
          "Column 2",
          "Column 3",
        ],
        "id": "1",
        "layout": 3,
        "partialScoring": Array [],
        "rows": Array [
          Object {
            "id": 1,
            "title": "Question Text 1",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 2,
            "title": "Question Text 2",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 3,
            "title": "Question Text 3",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 4,
            "title": "Question Text 4",
            "values": Array [
              false,
              false,
            ],
          },
        ],
      },
    }
  }
  onSessionChange={[MockFunction]}
  rationale="This is rationale"
  session={Object {}}
/>
`;

exports[`Main render snapshot with teacher Instructions 1`] = `
<Main
  classes={
    Object {
      "collapsible": "Main-collapsible-5",
      "main": "Main-main-2",
      "mainContainer": "Main-mainContainer-1",
      "prompt": "Main-prompt-4",
      "toggle": "Main-toggle-3",
    }
  }
  model={
    Object {
      "config": Object {
        "choiceMode": "radio",
        "element": "match-element",
        "feedback": Object {
          "correct": Object {
            "default": "Correct",
            "type": "none",
          },
          "incorrect": Object {
            "default": "Incorrect",
            "type": "none",
          },
          "partial": Object {
            "default": "Nearly",
            "type": "none",
          },
        },
        "headers": Array [
          "Column 1",
          "Column 2",
          "Column 3",
        ],
        "id": "1",
        "layout": 3,
        "partialScoring": Array [],
        "rows": Array [
          Object {
            "id": 1,
            "title": "Question Text 1",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 2,
            "title": "Question Text 2",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 3,
            "title": "Question Text 3",
            "values": Array [
              false,
              false,
            ],
          },
          Object {
            "id": 4,
            "title": "Question Text 4",
            "values": Array [
              false,
              false,
            ],
          },
        ],
      },
    }
  }
  onSessionChange={[MockFunction]}
  session={Object {}}
  teacherInstructions="These are teacher instructions"
/>
`;
